squid.gemspec

```diff
@@ -20,6 +20,9 @@ Gem::Specification.new do |spec|
   spec.executables   = spec.files.grep(%r{^exe/}) { |f| File.basename(f) }
   spec.require_paths = ['lib']
 
+  if Gem::Version.new(RUBY_VERSION) >= Gem::Version.new('3.1.0')
```

Comment:
I don't believe that rubygems supports conditional dependencies. Can you explain a bit what you're trying to achieve with this change?

Reply:
Ah, I didn't realize that. I've updated with an older minimum version that does not require ruby 2.5+.

As for the reason, the `matrix` gem is no longer a default gem in 3.1+. 

If you don't mind dropping support for ruby 2.4. https://rubygems.org/gems/matrix

Reply:
Your latest revision seems like a good solution to me. Thanks!